romancal/skycell/skymap.py

```diff
@@ -486,6 +579,40 @@ def pixel_scale(self) -> float:
         """degrees per pixel"""
         return self._skymap.pixel_scale
 
+    def skycells_at(self, radec: NDArray[float]) -> list[list[SkyCell]]:
+        """
+        skycells containing the given point(s)
+
+        Parameters
+        ----------
+        radec: NDArray[float]
+            right ascension and declination of coordinate(s)
+        """
+
+        radec = np.array(radec)
+        if radec.ndim == 1:
+            radec = np.expand_dims(radec, axis=0)
+
+        vectorpoints = np.stack(sgv.lonlat_to_vector(radec[:, 0], radec[:, 1]), axis=1)
+        if not isinstance(vectorpoints, np.ndarray):
+            vectorpoints = np.ndarray([vectorpoints])
+        vectorpoints = sgv.normalize_vector(vectorpoints)
+
+        return [
+            [
+                skycell
+                for skycell in (
+                    SkyCell(self.data["skycell_start"] + skycell_index)
+                    for skycell_index in skycell_indices
+                    if skycell_index != len(self.skycell_indices)
+                )
+                if skycell.polygon.contains_point(vectorpoints[point_index, :])
+            ]
+            for point_index, skycell_indices in enumerate(
+                self.skycells_kdtree.query_ball_point(vectorpoints, SkyCell.length)
+            )
+        ]
```

Comment:
I'll just note that I find these very long return statements hard to parse, despite the helpful indentation!  I wanted to check that This would get every skycell.  SkyCell.length is documented as being in degrees but I think it is is radians.
https://github.com/spacetelescope/spherical_geometry/blob/0c4696dc3f71130dff195b4d0018c768604b33a5/spherical_geometry/great_circle_arc.py#L198-L199
If I do
0.0018846729895155984 * 3600 * 180 / pi / 0.055 / sqrt(2)
I get 4997 pixels, where the difference from 5k is probably distortion---just to say that this doesn't look like this intends to include any margin.  But OTOH I think the kdtree has the centers of the skycells, so no margin would really be SkyCell.length / 2, so this is (inadvertently?) including a large factor of two margin?